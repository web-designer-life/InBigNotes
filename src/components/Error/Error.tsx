import React from 'react';
import { Link } from 'react-router-dom';
import { ROUTES } from '../../constants';
import { 
    Container, 
    Message, 
    BackLink,
} from './style';

export default class Error extends React.Component {
	render() {
        return (
            <Container>
                <Message>404 - Not Found!</Message>
                <Link to={ROUTES.HOME}>
                    <BackLink>Go home</BackLink>
                </Link>
            </Container>
        );
    };
};
