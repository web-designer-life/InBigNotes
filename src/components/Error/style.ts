import styled from 'styled-components';

const Container = styled.div`
	display: flex;
	justify-content: center;
	align-items: center;
	height: 100vh;
	flex-direction: column;
`;

const Message = styled.h1`
    margin: 10px;
`;

export {
    Container,
    Message,
}