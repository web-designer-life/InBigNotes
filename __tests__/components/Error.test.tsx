import React from 'react';
import { shallow } from 'enzyme';
import Error from '../../src/components/Error';

const errorProps = {
    navigateToPage: () => { },
};

describe('Error component', () => {
    it('should render Error component', () => {
        const component = shallow(<Error {...errorProps} />);

        expect(component).toMatchSnapshot();
    });
});
