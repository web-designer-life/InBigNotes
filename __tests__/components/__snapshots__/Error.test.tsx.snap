// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error component should render Error component 1`] = `
<styled.div>
  <styled.h1>
    404 - Not Found!
  </styled.h1>
  <Button
    color="green"
    onClick={[Function]}
    text="Go home"
    type="button"
  />
</styled.div>
`;
